### PR TITLE
Merge two sorted arrays C++ code 

### LINKED ISSUE #909: Merge without extra space
## 🚀 Feature

Given two sorted arrays arr1[] and arr2[] in non-decreasing order with size n and m. The task is to merge the two sorted arrays in place, i. e., we need to consider all n + m elements in sorted order, then we need to put first n elements of these sorted in the first array and remaining m elements in the second array.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

I want to add this as a part of CH20.

## Language

- C++: @rutujadhanawade 
- C: @ag278 
- Go:
- Java: @ritvij14 
- JavaScript:
- Python: @narayan2111

### PR DESCRIPTION
**Pull request by rutujadhanawade**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This is C++ code for merging two sorted arrays without using extra space. 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

 #909

### CODE REVIEW

**Comment by rutujadhanawade** [comment]
@siddharth25pandey @Kajol-Kumari @iamrajiv please review this.. As a part of CH20

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rutujadhanawade your code is same as this https://www.geeksforgeeks.org/efficiently-merging-two-sorted-arrays-with-o1-extra-space/

**Comment by rutujadhanawade** [comment]
@iamrajiv 
Actually I used that code for reference only. I wanted to add different file. I have added my code now. Please kindly review this again

File: C-Plus-Plus/sort/MERGE_SORT_WITHOUT_EXTRA_SPACE.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    
+	return 0;
+}
+
+/*
+
+Time complexity : NlogN (where N will be max of n and m)
+
+Sample output:
+enter totle no.of elements
```
Comment (iamrajiv) [inline C-Plus-Plus/sort/MERGE_SORT_WITHOUT_EXTRA_SPACE.cpp]: @rutujadhanawade update like this `Enter the total number of elements in array1 and array2 respectively`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rutujadhanawade Please format the code properly it has a lot of indentations. For reference, you can follow this https://google.github.io/styleguide/cppguide.html.

File: C-Plus-Plus/sort/MERGE_SORT_WITHOUT_EXTRA_SPACE.cpp (hunk 1 above)
Comment (rutujadhanawade) [reply C-Plus-Plus/sort/MERGE_SORT_WITHOUT_EXTRA_SPACE.cpp]: Done :+1:

**Comment by rutujadhanawade** [comment]
@iamrajiv please review this PR and add CH20 label to #909

**Comment by rutujadhanawade** [comment]
Hey @iamrajiv
 I have made the required changes 2 days ago , please review this.
@atarax665 @plazzy99 Please review this as a part of CH20

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rutujadhanawade Rename the file in lower case something like this `merge_sort_without_extra_space.cpp ` and move the program to `other` section.

**Comment by rutujadhanawade** [comment]
@iamrajiv anything else required?
 And actually the problem is I have created this PR from my master branch.  So I am not able to make new PR s so can I just close this PR and make new PR from a new branch so that I can add more PR related to different issues also?

**Comment by rutujadhanawade** [comment]
I will make sure to add the changes in new PR.

**Comment by iamrajiv** [comment]
@rutujadhanawade You do not have to make a new PR you can edit the file from the master branch only.

**Comment by rutujadhanawade** [comment]
Yeah But it will take some time to review my PR from each reviewer.. Till then I cant make new PR related to new issues. So should I do that?
@iamrajiv

**Comment by iamrajiv** [comment]
@rutujadhanawade Ok go ahead and make sure in future you make a separate branchs for separate issues.

**Comment by iamrajiv** [comment]
@rutujadhanawade Ok go ahead and make sure in future you make a separate branch for a separate issue.

**Pull request closed without merging**